singer-io/tap-mailchimp
# TDL-20258: Add custom exception handling

# Description of change
[TDL-20258](https://jira.talendforge.org/browse/TDL-20258): Add custom exception handling
- Added a custom exception class for each error code.
- Added backoff logic to retry MailchimpRateLimitError.

# Manual QA steps
 - Ran the sync mode and verify the error messages for every error code by raising errors manually.
 - Ran the sync mode and verify that tap retry for MailchimpRateLimitError 6 times.
 
# Risks
 - 
 
# Rollback steps
 - revert this branch
